radix-engine-macros/src/lib.rs

```diff
@@ -1,57 +1,37 @@
+use proc_macro::TokenStream;
+
+#[cfg(feature = "catch_unwind")]
 #[cfg(not(panic = "unwind"))]
 compile_error!("The `catch_unwind` crate requires that `panic = \"unwind\"` in the profile");
 
-use proc_macro::TokenStream;
-use proc_macro2::{Span, TokenStream as TokenStream2};
-use quote::{quote, ToTokens};
-use syn::*;
+#[cfg(feature = "catch_unwind")]
+mod unwind;
+#[cfg(feature = "catch_unwind")]
+use unwind::handle_catch_unwind;
 
+mod decimal;
+use decimal::{convert_to_i32, validate_decimal_string};
+
+#[cfg(feature = "catch_unwind")]
 #[proc_macro_attribute]
 pub fn catch_unwind(metadata: TokenStream, input: TokenStream) -> TokenStream {
     handle_catch_unwind(metadata.into(), input.into())
         .unwrap_or_else(|err| err.to_compile_error())
         .into()
 }
 
+#[cfg(feature = "catch_unwind")]
 #[proc_macro_attribute]
 pub fn ignore(_: TokenStream, input: TokenStream) -> TokenStream {
     input
 }
 
-fn handle_catch_unwind(metadata: TokenStream2, input: TokenStream2) -> Result<TokenStream2> {
-    let rtn_transformer = parse2::<Path>(metadata)?;
-
-    if let Ok(mut item_fn) = parse2::<ItemFn>(input.clone()) {
-        process_function(&mut item_fn.attrs, &mut item_fn.block, &rtn_transformer);
-        Ok(quote! { #item_fn })
-    } else if let Ok(mut item_impl) = parse2::<ItemImpl>(input) {
-        for item in item_impl.items.iter_mut() {
-            if let ImplItem::Method(method) = item {
-                process_function(&mut method.attrs, &mut method.block, &rtn_transformer)
-            }
-        }
-        Ok(quote! { #item_impl })
-    } else {
-        Err(Error::new(
-            Span::call_site(),
-            "Only functions and impls are supported by `catch_unwind`.",
-        ))
-    }
+#[proc_macro]
+pub fn decimal_string(input: TokenStream) -> TokenStream {
```

Comment:
Would prefer to keep all decimal related stuff within one crate `radix-engine-common`.
Unfortunately procedural macros require separate crate, therefore decided to reuse this crate.

Reply:
Comment no longer valid. Whole implementation refactored.